src/mono/mono/component/debugger-agent.c

```diff
@@ -5385,6 +5386,12 @@ decode_vtype (MonoType *t, MonoDomain *domain, gpointer void_addr, gpointer void
 		if (err != ERR_NONE)
 			return err;
 		nfields --;
+		if (CHECK_PROTOCOL_VERSION(2, 66) && inlineArray > 0)
```

Comment:
This is checking for `inlineArray > 0`, but the `for` loop uses only values from index `[1]` onwards.
Why is the first one skipped? And the condition should be changed to be `> 1` too?

Reply:
Changing to `i=0` does not break any tests. @thaystg, why?

Reply:
We are following the same thing that is done in the buffer_add_value_full.
The first element of the array is already sent here:
https://github.com/dotnet/runtime/blob/0c81f494f5664a462a26efdce0044875a22480f3/src/mono/mono/component/debugger-agent.c#L5269
Then we send the next the elements from 1 to array size.
And the same is done in the decode, on line 5385 we are decoding the first element, then we are decoding the next.

If you change it to i=0, and the first element has a value different of the second element you would see an error in the tests. I think!